src/Contracts/PhpSoap/Client.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+declare(strict_types=1);
+
+namespace RicorocksDigitalAgency\Soap\Contracts\PhpSoap;
+
+use SoapHeader;
+
+interface Client
+{
+    public function setEndpoint(string $endpoint): static;
```

Comment:
Not sure if the setters in this interface are a good idea though. See further down this code review - it allows for some strange structures.